src/main/java/com/fjb/sunrise/services/impl/UserServiceImpl.java

```diff
@@ -16,25 +17,33 @@
 public class UserServiceImpl implements UserService {
     @Value("${default.admin-create-key}")
     private String key;
+    private final UserMapper mapper;
     private final UserRepository repository;
     private final PasswordEncoder passwordEncoder;
 
     @Override
     public boolean checkRegister(RegisterRequest registerRequest) {
-        User user = new User();
-        user.setEmail(registerRequest.getEmail());
-        user.setFirstname(registerRequest.getFirstname());
-        user.setPhone(registerRequest.getPhone());
-        user.setLastname(registerRequest.getLastname());
+        //check already exist email -> unique email
+        if (repository.findByEmailOrPhone(registerRequest.getEmail()) != null) {
```

Comment:
Consider writing a query that checks for both email and phone to minimize database calls.